test/promql/tests/test_controller_reconciliation_errors.yaml

```diff
@@ -0,0 +1,127 @@
+        values: '1 2 3'
+
+    alert_rule_test:
+      - eval_time: 3m
+        alertname: ControllerReconciliationErrors
+        exp_alerts:
+          - exp_labels:
+              severity: warning
+              namespace: prod-ns2
+              pod: pod-2
+              controller: c2-controller
+              container: c2
+              service: s2
+            exp_annotations:
+              summary: "Controller c2-controller has reconciliation errors greater than 1%"
+              description: "Controller c2-controller with Pod pod-2 for namespace prod-ns2 has total reconciliation errors greater than 1% compared to total reconciliations in last 2 minutes."
+
+  - interval: 1m
+    input_series:
+      # Controller with large spikes in error values and pause in reconcile total,
+      # since the error rate is always greater than 1% in 2min window this is alerted
+      - series: 'controller_runtime_reconcile_errors_total{controller="c10-controller", namespace="prod-ns10", pod="pod-10", container="c10", service="s10"}'
+        values: '1 150 390'
+      - series: 'controller_runtime_reconcile_total{controller="c10-controller", namespace="prod-ns10", pod="pod-10", container="c10", service="s10"}'
+        values: '1 500 740'
+
+    alert_rule_test:
+      - eval_time: 3m
+        alertname: ControllerReconciliationErrors
+        exp_alerts:
+          - exp_labels:
+              severity: warning
+              namespace: prod-ns10
+              pod: pod-10
+              controller: c10-controller
+              container: c10
+              service: s10
+            exp_annotations:
+              summary: "Controller c10-controller has reconciliation errors greater than 1%"
+              description: "Controller c10-controller with Pod pod-10 for namespace prod-ns10 has total reconciliation errors greater than 1% compared to total reconciliations in last 2 minutes."
+
+  # Not Alerted test cases
+  - interval: 1m
+    input_series:
+      # Controller with no reconciliation events throughout the test time frame so this is not alerted
+      - series: 'controller_runtime_reconcile_errors_total{controller="c5-controller", namespace="prod-ns5", pod="pod-5", container="c1", service="s1"}'
+        values: '1 1 1'
+      - series: 'controller_runtime_reconcile_total{controller="c5-controller", namespace="prod-ns5", pod="pod-5", container="c1", service="s1"}'
+        values: '10 10 10'
+
+      # Controller with errors during parts of the test period
+      - series: 'controller_runtime_reconcile_errors_total{controller="c3-controller", namespace="prod-ns3", pod="pod-3", container="c1", service="s1"}'
+        values: '0 2 2'
+      - series: 'controller_runtime_reconcile_total{controller="c3-controller", namespace="prod-ns3", pod="pod-3", container="c1", service="s1"}'
+        values: '1 3 3'
```

Comment:
Why does this not raise the alert if during the first two minutes the increase in errors vs total is 2/2 or 100%?

Reply:
The series starts at 0m so at 2m the increase is 0.